### PR TITLE
Update MS SQL Server JDBC driver

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Update the JDBC driver for the SQL Server connector to the new major release (11) and also use the Java 17 version instead of the Java 11 version. The new driver is fully 4.2. JDBC spec compliant among other improvements.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

SQL Server connector only.

> How would you describe this change to a non-technical end user or system administrator?

Update to new JDBC driver mostly for maintenance reasons. 

## Related issues, pull requests, and links

* Release notes https://docs.microsoft.com/en-us/sql/connect/jdbc/release-notes-for-the-jdbc-driver?view=sql-server-ver16#112

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
## SQL Server connector
* to be determined ({issue}`13948`)
```

There are a bunch of fixes and improvements in the JDBC driver that could affect Trino usage but I am not sure there are. Some of the things supported by the driver like IPv6 config and such can potentially be used with connection string parameters but I am uncertain what to mention in the release notes for Trino. 

E.g. you can now enable TDS 8.0 support when the property `encrypt` is set to `strict`, which in fact is a breaking change for users that used that property before.

### CODE REVIEW

**Comment by mosabua** [comment]
I filed this as WIP for now to verify passing test suite while checking for other necessary updates.

File: pom.xml
Comment (hashhar) [inline pom.xml:1228]: It's a very recent release. IMO we should wait for at least 1 more release to make sure bugs are caught. In the past we've had some cases where regressions were found due to driver bugs.

Also Java is backward compatible so a JDK 11 lib (which doesn't do illegal things) should keep working.

Comment (mosabua) [reply pom.xml:1228]: Sure.. we can wait.

Also I realize that the JDK 11 version will continue to work but the 17 version will use better optimized bytecode and hence be better to use. We do require Java 17 after all ;-)

Comment (hashhar) [reply pom.xml:1228]: I think so.

Comment (mosabua) [reply pom.xml]: Done

**Comment by mosabua** [comment]
Once CI passes this is ready for merge imho .. however I will need help to determine if a RN entry is needed, and if yes.. what it should say.

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
For RN I don't anything user-facing. So we can either not mention it or just mention driver version is updated. What did we do in past?

**Comment by mosabua** [comment]
@hashhar last time there were using facing breakages so we documented. In this case there are as well .. but super edge usage..

**Comment by mosabua** [comment]
@ebyhr @hashhar @martint .. merge maybe?

**Comment by hashhar** [comment]
There is:

> TDS 8.0 support by adding "strict" option to encrypt connection property | BREAKING CHANGE Encrypt connection property is now of type string.

But it's not breaking from Trino's perspective since we don't set it. However it would be good idea to mention in release notes that the driver version is updated (and link to release notes) so people can do any changes that they need for their setup.

**Comment by mosabua** [comment]
Thanks for the merge @hashhar

**Pull request merged**
